src/charm.py

```diff
@@ -183,73 +223,31 @@ def _set_mimir_config(self) -> bool:
             logger.error("Failed to push updated config file: %s", e)
             raise BlockedStatusError(str(e))
 
-    def _set_alerts(self) -> None:
-        """(Re-)create alert rule files for all Mimir consumers.
-
-        Returns: True if alerts rules have changed, otherwise False.
-        Raises: BlockedStatusError exception if PebbleError, ProtocolError, PathError exceptions
-            are raised by container.remove_path
-        """
-        try:
-            self._container.remove_path(RULES_DIR, recursive=True)
-        except PebbleError as e:
-            logger.error("Failed to remove alerts directory: %s", e)
-            raise BlockedStatusError("Failed to remove alerts directory; see debug logs")
-
-        try:
-            self._push_alert_rules(self.remote_write_provider.alerts())
-        except (ProtocolError, PathError) as e:
-            logger.error("Failed to push updated alert files: %s", e)
-            raise BlockedStatusError("Failed to push updated alert files; see debug logs")
-
-    def _push_alert_rules(self, alerts):
-        """Push alert rules from a rules file to the mimir container.
-
-        Args:
-            alerts: a dictionary of alert rule files.
-        """
-        # Without multitenancy, the default is `anonymous`, and the ruler checks under
-        # {RULES_DIR}/<tenant_id>
-        tenant_dir = f"{RULES_DIR}/anonymous"
-        # Need to `make_dir` even though we have `make_dirs=True` below
-        # https://github.com/canonical/operator/issues/898
-        self._container.make_dir(tenant_dir, make_parents=True)
-        for topology_identifier, rules_file in alerts.items():
-            filename = f"juju_{topology_identifier}.rules"
-            path = os.path.join(tenant_dir, filename)
-            rules = yaml.safe_dump(rules_file)
-            self._container.push(path, rules, make_dirs=True)
-            logger.debug("Updated alert rules file %s/%s", path, filename)
-
-    @property
-    def _pebble_layer(self):
-        return Layer(
-            {
-                "summary": "mimir layer",
-                "description": "pebble config layer for mimir",
-                "services": {
-                    "mimir": {
-                        "override": "replace",
-                        "summary": "mimir daemon",
-                        "command": f"/bin/mimir --config.file={MIMIR_CONFIG}",
-                        "startup": "enabled",
-                    }
+    def _build_mimir_config(self) -> dict:
+        if s3 := self.object_storage.bucket_info:
+            common_storage = {
+                "backend": "s3",
```

Comment:
What will happen to mimir/data if after `s3 broken` it is restarted with a different config (filesystem instead of s3)? Or vice versa?
I.e. do we need to manually migrate anything?

Reply:
The data is lost. "don't do that". Mimir has tools to dump and load, but they're not covered here. I also don't expect them to be. 

Note that loki, tempo, and phlare will have identical scenarios and problems. It's just a new thing for us.

Reply:
Also note the data presumably isn't lost. It is just no longer accessible to mimir.

Reply:
I think this should be documented (README?) as it is very easy to lose (access to) data this way unintentionally.

Reply:
I strongly agree with Leon that this needs to be documented. Preferably we would also log whenever we detect potential data abandonment.